src/zenml/cli/stack_components.py

```diff
@@ -228,6 +237,14 @@ def register_stack_component_command(
             list(args) + [name], expand_args=True
         )
 
+        parsed_metadata = None
+        # split metadata by "=" and create a dict
+        if metadata:
+            parsed_metadata = {}
+            for m in metadata:
+                key, value = m.split("=")
+                parsed_metadata[key] = value
+
```

Comment:
This probably could be extracted into a utils function

Reply:
Good catch, I've added it as a fn.